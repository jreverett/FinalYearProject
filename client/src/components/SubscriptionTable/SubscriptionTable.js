import React, { Component } from 'react';
import { SubscriptionRow } from '..';
import '../../common.css';
import './SubscriptionTable.css';

class SubscriptionTable extends Component {
  render() {
    var subscriptions = this.props.loggedInUser.subscriptions;
    var subscriptionRows = [];

    if (subscriptions) {
      subscriptionRows = subscriptions.map((sub, index) => {
        return (
          <SubscriptionRow
            key={index}
            userID={this.props.loggedInUser._id}
            eventID={sub}
          />
        );
      });
    }

    return (
      <div className="table-container">
        <div>{subscriptionRows}</div>
      </div>
    );
  }
}

export default SubscriptionTable;
