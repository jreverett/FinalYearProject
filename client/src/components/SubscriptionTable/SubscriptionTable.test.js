import React from 'react';
import renderer from 'react-test-renderer';
import SubscriptionTable from './SubscriptionTable';

describe('<SubscriptionTable />', () => {
  it('matches the snapshot', () => {
    const testUser = { ownedEvents: [] };
    const tree = renderer
      .create(<SubscriptionTable loggedInUser={testUser} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
