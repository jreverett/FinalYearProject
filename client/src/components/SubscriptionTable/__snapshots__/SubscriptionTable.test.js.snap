// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SubscriptionTable /> matches the snapshot 1`] = `
<div
  className="table-container"
>
  <div />
</div>
`;
